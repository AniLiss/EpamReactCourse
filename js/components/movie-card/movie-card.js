import React from 'react';

export const MovieCard = ({title, releaseDate, genre, key}) => {
    return (
        <article className="movie-card"  key={key}>
            <img className="movie-card__img" alt="Kill Bill. Vol. 1" />
            <div className="movie-card__desc">
                <h3 className="movie-card__title">{title}</h3>
                <span className="movie-card__year">{releaseDate}</span>
                <span className="movie-card__genre">{genre}</span>
            </div>
        </article>
    )
};
