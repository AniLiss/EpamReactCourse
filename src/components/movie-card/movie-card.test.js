import React from 'react';
import {shallow} from 'enzyme';
import {MovieCard} from './movie-card';

const wrapper = shallow(
    <MovieCard
        title='Title'
        releaseDate='02/03/1994'
        genre='drama'
    />
);

describe('MovieCard', () => {
    it('is expected to render MovieCard correctly', () => {
       expect(wrapper).toMatchSnapshot();
       expect(wrapper.find('article').exists()).toBe(true);
    })
});
