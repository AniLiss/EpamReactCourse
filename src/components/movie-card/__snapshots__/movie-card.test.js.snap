// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard is expected to render MovieCard correctly 1`] = `
<article
  className="movie-card"
>
  <img
    alt="Kill Bill. Vol. 1"
    className="movie-card__img"
  />
  <div
    className="movie-card__desc"
  >
    <h3
      className="movie-card__title"
    >
      Title
    </h3>
    <span
      className="movie-card__year"
    >
      02/03/1994
    </span>
    <span
      className="movie-card__genre"
    >
      drama
    </span>
  </div>
</article>
`;
